projects/ngx-d3timeline/src/lib/scale-band.service.ts

```diff
@@ -0,0 +1,28 @@
+import { scaleBand, ScaleBand } from 'd3-scale';
+import { TimelineEvent } from './timeline-event';
+import { TimelineView } from './view/timeline-view';
+import { Orientation } from './orientation';
+import { Injectable } from '@angular/core';
+
+@Injectable({ providedIn: 'root' })
+export class ScaleBandService {
+  configureScaleBand(
+    data: TimelineEvent[],
+    timelineView: TimelineView,
+    orientation: Orientation
+  ): ScaleBand<string> {
+    return scaleBand()
+      .domain(this.getScaleBandDomain(data))
+      .range(this.getRange(timelineView.bounds, orientation));
+  }
+
+  private getScaleBandDomain(data: TimelineEvent[]): string[] {
```

Comment:
don't need the ScaleBand prefix now given this is the ScaleBandService + getRange doesn't have a prefix

```suggestion
  private getDomain(data: TimelineEvent[]): string[] {
```

Reply:
done